resources/sdk/purecloudjava/extensions/extensions/notifications/NotificationHandler.java

```diff
@@ -336,4 +264,84 @@ protected void finalize() throws Throwable {
         }
         super.finalize();
     }
+
+    private WebSocket createWebSocket() throws IOException {
+        WebSocketFactory factory = new WebSocketFactory();
+
+        if (this.proxyHost != null)
+            factory.getProxySettings().setServer(this.proxyHost);
+
+        return factory
+                .createSocket(this.channel.getConnectUri())
+                .addListener(new WebSocketAdapter() {
+                    @Override
+                    public void onStateChanged(WebSocket websocket, WebSocketState newState) throws Exception {
```

Comment:
Small change.  Embedding the class definition directly in the addListener call makes the code a little harder to follow.  Could you break this class into a small inner class definition and then pass an instance of the class into the addListener class.  It would make things easier to read and follow.